src/main/java/com/spotify/folsom/ketama/KetamaMemcacheClient.java

```diff
@@ -89,41 +89,23 @@ private RawMemcacheClient getClient(final byte[] key) {
       routing2.add(client);
     }
 
-    final Map<RawMemcacheClient, ListenableFuture<List<T>>> futures = Maps.newIdentityHashMap();
+    final Map<RawMemcacheClient, CompletableFuture<List<T>>> futures = Maps.newIdentityHashMap();
 
     for (final Map.Entry<RawMemcacheClient, List<byte[]>> entry : routing.entrySet()) {
       final List<byte[]> subKeys = entry.getValue();
       final Request<List<T>> subRequest = multiRequest.create(subKeys);
       final RawMemcacheClient client = entry.getKey();
-      ListenableFuture<List<T>> send = client.send(subRequest);
+      CompletableFuture<List<T>> send = client.send(subRequest);
       futures.put(client, send);
     }
-    final ListenableFuture<List<List<T>>> allFutures = Futures.allAsList(futures.values());
-    return Utils.transform(allFutures, new Assembler<>(futures, routing2));
-  }
-
-  private static class Assembler<T, R> implements Function<List<List<T>>, List<T>> {
```

Comment:
Did I completely miss the point here, is the assembler doing anything more than just merging the result?

Reply:
It merges it, but in the correct order.